### PR TITLE
feat: Reduce templating dependency on devops/main.tf

### LINKED ISSUE #981: Remove main.tf templating logic from Devops component
While generating `variables.tf` file transferred some of the major logic into the `terraform.tfvars` file, there were some /conditional resources and fields that are still dependant on templating.

As stated at the discussion of #957, there were 3 main problems with the complete reduction of templating logic:

1. **Conditional resources**: Some resources turned out to be dependant on the existence of some variables. This conditional can be reproduced with the [`count` meta argument](https://www.terraform.io/docs/language/meta-arguments/count.html)
2. **Conditional nested fields**: Conditional nested fields were kept with templated `if` statements due to the restriction that terraform has of omitting nested fields. This can be solved by providing default templated values on `.tfvars` that are the defaults on the underlying terraform modules (every module has a default.). Passing the underlying default is equivalent to not pass anything at all and it avoids the error caused by the nested variable omission.
3. **Descriptions for nested values**: Nested values were not with a description provided, but now, with #953, descriptions are now more maintainable since they're added as variables to the templating, and, now tested against internal resources, I'm proposing a format to show nested descriptions like the following:

```
variable "foo" {
  type = object({
    nestedFoo: object({
      nestedSubFoo: ...,
    }),
    nestedBar: ...
   })

  description = <<EOF
    {{props.foo.description}}

   * nestedFoo = {{...props.nestedFoo.description}}
   ** nestedSubFoo = {{...props.nestedSubFoo.description}}
   * nestedBar = {{...props.nestedFoo.description}}
  EOF
}
```

Applying these 3 principles to devops recipe, templating can be theoretically removed

### PR DESCRIPTION
**Pull request by ernestognw**
Closes #981

### CODE REVIEW

File: templates/tfengine/components/devops/terraform.tfvars
Comment (MartinPetkov) [inline templates/tfengine/components/devops/terraform.tfvars:23]: For style, I think the one-liner version here is less readable than the expanded multi-line version, I would prefer you use the latter.

File: templates/tfengine/components/devops/variables.tf
Comment (MartinPetkov) [inline templates/tfengine/components/devops/variables.tf:32]: Same as the other PR, consider just looping through `$adminsGroupProps` here and adding a "Fields:" line right before to make it clearer what this section of the description is.

Comment (ernestognw) [reply templates/tfengine/components/devops/variables.tf:34]: Same resolution as the other PR, only added Field header

File: templates/tfengine/components/devops/terraform.tfvars
Comment (ernestognw) [reply templates/tfengine/components/devops/terraform.tfvars:23]: You're right, I was kinda unsure wether to use this one or the multiline. For empty cases, the multiline ends up being like:

```
owners = [
]
```

I changed it to multiline

File: examples/tfengine/generated/org_foundation/devops/main.tf
Comment (xingao267) [inline examples/tfengine/generated/org_foundation/devops/main.tf:71]: the logic is opposite.

count should 1 when the group does not exist

File: examples/tfengine/generated/org_foundation/devops/variables.tf
Comment (xingao267) [inline examples/tfengine/generated/org_foundation/devops/variables.tf:37]: what about managers and members?

File: templates/tfengine/components/devops/variables.tf
Comment (xingao267) [inline templates/tfengine/components/devops/variables.tf:94]: extra s at the end

File: templates/tfengine/components/devops/variables.tf
Comment (xingao267) [inline templates/tfengine/components/devops/variables.tf:93]: extra s at the end

File: templates/tfengine/components/devops/variables.tf
Comment (xingao267) [inline templates/tfengine/components/devops/variables.tf:91]: Can we use extra indent instead of double *?

File: examples/tfengine/generated/team/devops/terraform.tfvars
Comment (xingao267) [inline examples/tfengine/generated/team/devops/terraform.tfvars:43]: Can this be formatted as `owners = []` ?

Comment (ernestognw) [reply examples/tfengine/generated/team/devops/terraform.tfvars:43]: I mentioned here https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/pull/984#discussion_r682769050

But I think that's preferrable. Other functions like hcl write the emails without the "

```
owners = [user@example.com]
```

instead of

```
owners = ["user@example.com"]
```

I didn't found other way around

File: templates/tfengine/components/devops/variables.tf
Comment (ernestognw) [reply templates/tfengine/components/devops/variables.tf:91]: We can, but that might add extra complexity if we automate this by iterating the properties as Martin suggested in other PRs.

What do you think might be the best?

File: examples/tfengine/generated/org_foundation/devops/variables.tf
Comment (ernestognw) [reply examples/tfengine/generated/org_foundation/devops/variables.tf:37]: They're disabled at the schema, we agreed on the first PR to keep them in templates but leaving them commented on the schema.

I cannot access the description if it's disabled

File: examples/tfengine/generated/org_foundation/devops/main.tf
Comment (ernestognw) [reply examples/tfengine/generated/org_foundation/devops/main.tf]: That's right. Thanks!

File: templates/tfengine/components/devops/variables.tf
Comment (MartinPetkov) [reply templates/tfengine/components/devops/variables.tf]: Maybe this should be part of the schemaDescription function, to also output the internal field descriptions. Then you could iterate and use recursion (for nested objects) with normal code instead of templating code. Would that be complex to do?

Comment (ernestognw) [reply templates/tfengine/components/devops/variables.tf]: Sounds doable and better, but the problems are almost the same:

1. Sometimes properties are inside properties key or items key (this is fixable if I understand better when it's one or the other)
2. Some schema variables may be renamed into the `variables.tf` or don't appear at all. How can we detect those?

Comment (MartinPetkov) [reply templates/tfengine/components/devops/variables.tf]: Good point.

Then in the interest of not debating this forever, I'm okay with not iterating for now. Besides that, I agree that an extra indent would look better than 2 **s

File: examples/tfengine/generated/org_foundation/devops/variables.tf
Comment (xingao267) [reply examples/tfengine/generated/org_foundation/devops/variables.tf:37]: oh right. my bad.

File: examples/tfengine/generated/team/devops/terraform.tfvars
Comment (xingao267) [reply examples/tfengine/generated/team/devops/terraform.tfvars:43]: I actually mean removing the line break if it's an empty list, 

like [], instead of [
]


but it's very minor.

File: templates/tfengine/components/devops/variables.tf
Comment (ernestognw) [reply templates/tfengine/components/devops/variables.tf]: Got it. Looks better, thanks!

**Review by MartinPetkov — COMMENTED** [review]
LGTM but I'll let Xin approve

File: examples/tfengine/generated/team/devops/terraform.tfvars
Comment (ernestognw) [reply examples/tfengine/generated/team/devops/terraform.tfvars:43]: Resolved in favor of https://github.com/GoogleCloudPlatform/healthcare-data-protection-suite/pull/984#:~:text=I%20mentioned%20here-,%23984%20(comment),-But%20I%20think

File: templates/tfengine/components/devops/main.tf
Comment (xingao267) [inline templates/tfengine/components/devops/main.tf:131]: 0 : 1

File: templates/tfengine/components/devops/terraform.tfvars
Comment (xingao267) [inline templates/tfengine/components/devops/terraform.tfvars:22]: one extra space before the `=`.

Also format this in the same way as the below owners_group. I think the owners_group one looks a bit better.

File: templates/tfengine/components/devops/main.tf
Comment (ernestognw) [reply templates/tfengine/components/devops/main.tf]: You're right. Thanks!

File: examples/tfengine/generated/multi_envs/devops/terraform.tfvars
Comment (xingao267) [inline examples/tfengine/generated/multi_envs/devops/terraform.tfvars:38]: This might not work with. The `,` at the end might cause issue

File: templates/tfengine/components/devops/terraform.tfvars
Comment (ernestognw) [reply templates/tfengine/components/devops/terraform.tfvars]: Done, I set it as the owners_group

File: examples/tfengine/generated/multi_envs/devops/terraform.tfvars
Comment (ernestognw) [reply examples/tfengine/generated/multi_envs/devops/terraform.tfvars:42]: Tested with these 3 different combinations on a cicd recipe. They all work.

I'm resolving

Screenshot 2021-08-11 11 57 03 AM
Screenshot 2021-08-11 11 59 54 AM
Screenshot 2021-08-11 12 00 38 PM

**Review by xingao267 — APPROVED** [review]

**Comment by xingao267** [comment]
Thanks. LGTM. There is some merging conflicts. Could you take a look and once they are resolved, i'll merge it.

**Review by MartinPetkov — APPROVED** [review]

**Review by xingao267 — APPROVED** [review]

**Pull request merged**
